lib/active_record/connection_adapters/abstract_mysql.rb

```diff
@@ -1,21 +1,19 @@
-# This module fails in Rails 7.0 becuase the method being modified has been
-# changed to a class method.
-
 module ActiveRecord
   module ConnectionAdapters
     class AbstractMysqlAdapter
```

Comment:
I think this is reopening the original adapter, is that right? Is that because there are effectively no hooks we could use? Or would this potentially be a good time to move to a proper initializer that hooks the adapter at that point?

Reply:
I wonder if maybe the thing to do is keep the `register_enum_type` method here (there are other, similar methods in the adapter already) and move the call of it to a Railtie or something

Reply:
My inclination is to get this PR in and then try to mess with that separately?